### PR TITLE
pre-workshop survey questions to probe learner proficiency

### LINKED ISSUE #82: Pre-workshop survey questions
I've been trying to come up with a set of questions to ask learners what material from this lesson they know already. I'm planning on using these with the trial run next week. I've tried to cover the whole range of topics we're currently using but might have missed some topics.  My current questions are: 

> 1. Describe your previous machine learning experience (if any).
> 2. Which of the following can you do already? Answer Yes, Maybe or No for each (don't worry if you can't do any of these):
> Decide if a problem might be solved using deep learning.
> Prepare data so that it can be used to train a neural network.
> Apply a pre-trained model to a dataset.
> Build your own multi-layer neural network using a library such as Keras, PyTorch or Tensorflow.
> Add convolutional or pooling layers to a neural network using a library such as Keras, PyTorch or Tensorflow.
> Use a confusion matrix to evaluate a classifier's accuracy.
> Identify when a network is overfitting its training data.

Does anyone have any suggestions for improvements on these?

### PR DESCRIPTION
**Pull request by psteinb**
Closes #82

### CODE REVIEW

**Comment by psteinb** [comment]
@colinsauze As this PR targets an issue you created, I take the liberty of kindly ask a review of you, please. ;-)

**Comment by psteinb** [comment]
Uaggh, silly me. This PR doublicates #59 to some extent. What should we do?

**Comment by colinsauze** [comment]
> Uaggh, silly me. This PR doublicates #59 to some extent. What should we do?

Do we consider this one to be more developed and the one we should now work from?

**Comment by psteinb** [comment]
That is the central question, we need to find an answer for.

**Comment by colinsauze** [comment]
If there's no major difference perhaps we should use #59 since it has a lot of discussion already.

File: _extras/survey-templates.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+- You are helping to organize a conference of more than 1000 attendants. All participants have already paid and are expecting to pick up their conference t-shirt on the first day. Your team is in shock as it discovers that t-shirt sizes have not been recorded during online registration. However, all participants were asked to provide their age, gender, body height and weight. To help out, you sit down to write a python script that predicts the t-shirt size for each participant using a clustering algorithm.
+
+- You are presented a training data set of 10000 samples. You would like to train a classifier on this datasets. However, you observe that class 0 dominates the training set at 60%, the other classes equally share 20% of the rows each. How do you prepare your training procedure?
+
+-  You are given a classifier by a colleague. Your peer trained it on a very large dataset. You just need it for inference. You reassure yourself if the model you loaded from disk really reproduces the weights by running the 'predict' function on an unseen validation set by your colleague. The network doesn't make the same predictions as your colleague told you. What could be the cause?
```
[the remark this answers is not in the document]
Comment (elcorto) [reply _extras/survey-templates.md:71]: "The ~network~ model doesn't make the same predictions as your colleague told you." One could specify which data set
the colleague used to create predictions that now cannot be reproduced.

File: _extras/survey-templates.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+### Questions about python coding
+
+**The following questions range from simple to more advanced.**
+
+- You are provided with a python list of integer values. The list has length 1024 and you would like to obtain all entries from index 50 to 101. How would you do this?
+
+- You need to open 102 data files and extract an object from them. For this, you compose a small function to open a single file which requires 3 input parameters. The parameters are a file location, the name of the object to retrieve and a parameter that controls the verbosity of the function. The latter parameter has the default value “False”.
+
+- Consider a 4 dimensional numpy array of uint32 type numbers. The shape of the array is '(512,3,224,224)' and it is stored in row-major memory ordering. Your goal is to randomly set every other row to 42, i.e. every other entry in dimension 2 to 42.
```
[the remark this answers is not in the document]
Comment (psteinb) [reply _extras/survey-templates.md:39]: true, the `randomly` needs to go.

[the remark this answers is not in the document]
Comment (psteinb) [reply _extras/survey-templates.md]: fixed with https://github.com/psteinb/deep-learning-intro/commit/cc84efded86d166ba3c0df6568e37b5b3380e349

File: _extras/survey-templates.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+- You are provided with a CSV file. The file has 35000 rows. The file has 45 columns. Each column is separated by the “,” symbol from the other. You would like to open the file in python, remove all entries where the value of column 21 is larger than 50. The values removed are to be replaced by 0.
+
+- You are provided with a CSV file. The file has 35000 rows. The file has 45 columns. Each column is separated by the “,” symbol from the other. When you load the file and plot the histogram of column 40, you are suspicious that the floating point values are not normally distributed. But, the producer of the CSV file assures you that all columns are normally distributed. To make sure, you sit down to code a function which tests any given column if it is normally distributed.
+
+- A probability density function (PDF) in statistics is something very distinct from a probability mass function (PMF). What is the difference?
+
+- You are given samples of two sets of observations with type float32. Both arrays have a similar shape: "(1024, 1)" You'd like to compare both ensembles if they are produced by the same PDF. How do you do this?
```
[the remark this answers is not in the document]
Comment (psteinb) [reply _extras/survey-templates.md:59]: well spotted.

File: _extras/survey-templates.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+**The following questions range from simple to more advanced.**
+
+- You are provided with a python list of integer values. The list has length 1024 and you would like to obtain all entries from index 50 to 101. How would you do this?
+
+- You need to open 102 data files and extract an object from them. For this, you compose a small function to open a single file which requires 3 input parameters. The parameters are a file location, the name of the object to retrieve and a parameter that controls the verbosity of the function. The latter parameter has the default value “False”.
+
+- Consider a 4 dimensional numpy array of uint32 type numbers. The shape of the array is '(512,3,224,224)' and it is stored in row-major memory ordering. Your goal is to randomly set every other row to 42, i.e. every other entry in dimension 2 to 42. 
+
+- You discovered the numpy 'rot90' method. You observe yourself calling it over and over again with the same parameter for the axis option. You'd like to write a wrapper function which calls 'rot90' having axis set to '(0,1)'.
```
[the remark this answers is not in the document]
Comment (psteinb) [reply _extras/survey-templates.md:41]: Not really, this question is about composing a wrapper function.

File: _extras/survey-templates.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (psteinb) [reply _extras/survey-templates.md]: fixed with a43608b

File: _extras/survey-templates.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (psteinb) [reply _extras/survey-templates.md]: I adapted the wording a bit. 9f93d49 
What do you think?

I'd not be over-precise on the dataset question you raised so that people are not too confused. Or maybe I mistake what you mean. Could you provide an example, please?

[the remark this answers is not in the document]
Comment (elcorto) [reply _extras/survey-templates.md]: Hmm, I understood:

* colleague trains model unsing `X_train`, `Y_train`
* tells me "result should be `Y_test`", doesn't tell me input is `X_test`
* I use `Y_valid = model.predict(X_valid) != Y_test`
